### PR TITLE
 Implement deployment API support for ant-man and flash previews

### PR DESCRIPTION
**Pull request by marcoferrer**
# Description 
* Implement deployment API support for ant-man and flash previews.
* This pr includes two new constants in the `Previews` class. 
* Getters for newly introduced fields are added to `GHDeployment` and `GHDeploymentStatus`.
* New builder setters are implemented for `GHDeploymentBuilder` and `GHDeploymentStatusBuilder`.
* `GHDeploymentState` has been updated to support `in_progress`, `queued`, and `inactive`. 
  * [Related Gh API Docs](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-deployment-status)
* Marked `GHDeploymentStatus.getTargetUrl()` and `GHDeploymentStatusBuilder.targetUrl()` as `@Deprecated`.
  * Method documentation has been updated to advise users to replace usages with `logUrl()` where appropriate

### Github Blog Posts For Previews Implemented
* [flash](https://developer.github.com/changes/2018-10-16-deployments-environments-states-and-auto-inactive-updates/) _(10/16/2018)_
* [ant-man](https://developer.github.com/changes/2016-04-06-deployment-and-deployment-status-enhancements/) _(04/06/2016)_
# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [X] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [X] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [X] Fill in the "Description" above. 
- [X] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by marcoferrer** [comment]
@bitwiseman Tests are implemented and this is ready for review. One change that I wanted feedback on was the addition of a value field to the `Preview` annotation. Thx

**Comment by marcoferrer** [comment]
@bitwiseman friendly bump

File: src/main/java/org/kohsuke/github/Preview.java [hunk 1]
```diff
@@ -15,4 +15,15 @@
...
+    /**
+     * An optional field defining what API media types must be set inorder to support the usage of this annotations
+     * target.
+     * <p>
+     * This value should be set using the existing constants defined in {@link Previews}
+     *
+     * @return The API preview media type.
+     */
+    public String[] value() default {};
+
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/Preview.java:28]: Very nice.  I'd almost prefer it to not be optional but I see why it needs to be.

File: src/main/java/org/kohsuke/github/GHDeployment.java [hunk 2]
```diff
@@ -89,6 +92,16 @@ public Object getPayloadObject() {
         return payload;
     }
 
+    /**
+     * The environment defined when the deployment was first created.
+     *
+     * @return the original deployment environment
+     */
+    @Preview(Previews.FLASH)
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHDeployment.java:100]: We mark preview fields as `@Deprecated` to make sure end users are aware (via editor feedback) that they are not stable APIs. 

This comment applies to all usages. 

```suggestion
    @Deprecated
    @Preview(Previews.FLASH)
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Sorry for long delay. 

Only one type of change needed.  Otherwise looks great.

**Comment by marcoferrer** [comment]
@bitwiseman Feedback addressed. `@Deprecated` has been added to applicable targets.

File: src/main/java/org/kohsuke/github/Preview.java [hunk 3]
```diff
@@ -8,11 +8,23 @@
  * Indicates that the method/class/etc marked maps to GitHub API in the preview period.
  * <p>
  * These APIs are subject to change and not a part of the backward compatibility commitment. Always used in conjunction
- * with 'deprecated' to raise awareness to clients.
+ * with 'deprecated' to raise awareness to clients. In addition, it's advised to update the targets documentation to
+ * signify that the deprecation is required until preview feature being used is promoted to stable.
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/Preview.java:12]: 👍

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**